dedupe/ecr_to_fhir_conversion/ECR to FHIR setup.ipynb

```diff
@@ -0,0 +1,196 @@
+   "cell_type": "markdown",
+   "id": "3e95940d-9fcd-4629-8660-2389a5db00fd",
+   "metadata": {},
+   "source": [
+    "#### 2. For each zip file in the archive, extract all of the other files into a folder named after the zip file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "6e846502-92c4-4b98-bcbd-86a094f2feab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_files = list(outfile.glob(\"*.*\"))\n",
+    "for f in all_files:\n",
+    "    with ZipFile(f, \"r\") as zip_file:\n",
+    "        # assumes you are in the phdi directory\n",
+    "        zip_file.extractall(path=f\"./containers/ecr-viewer/seed-scripts/baseECR/{f.stem}/\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c47e3df2-631b-4ce6-aa11-f390ad783ef2",
+   "metadata": {},
+   "source": [
+    "#### 3. Launch FHIR Conversion Docker container\n",
+    "\n",
+    "Navigate to the `ecr-viewer` subdirectory in the `phdi` directory via `cd containers/ecr-viewer` in your terminal and run the following commands:\n",
+    "\n",
+    "```bash\n",
+    "dockId=$(docker run --rm -d -it -p 8080:8080 \"$(docker build -q ./../fhir-converter/)\")\n",
+    "echo $dockId\n",
+    "```\n",
+    "\n",
+    "As long as you see a value printed to the screen, your container should be up and running."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b3f043ac-4bac-4a1a-9f30-efe4f7b1b0b2",
+   "metadata": {},
+   "source": [
+    "#### 4. Loop through each folder and extract the files\n",
+    "\n",
+    "Run the following for loop in your terminal:\n",
+    "\n",
+    "```bash\n",
+    "for d in ./baseECR/* ; do\n",
+    "    #first escape \", then /, and finally remove all new lines\n",
+    "    rr=$(sed -e 's/\"/\\\\\"/g ; s=/=\\\\\\/=g ; $!s/$/\\\\n/' \"$d/CDA_RR.xml\" | tr -d '\\n')\n",
+    "    eicr=$(sed 's/\"/\\\\\"/g ; s=/=\\\\\\/=g ; $!s/$/\\\\n/'  \"$d/CDA_eICR.xml\" | tr -d '\\n')\n",
+    "    resp=$(curl -l 'http://localhost:8080/convert-to-fhir' --header 'Content-Type: application/json' --data-raw '{\"input_type\":\"ecr\",\"root_template\":\"EICR\",\"input_data\": \"'\"$eicr\"'\",\"rr_data\": \"'\"$rr\"'\"}')\n",
+    "    echo $resp | jq '.response.FhirResource' > \"./fhir_data/$(basename $d).json\"\n",
+    "done\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "306d7319-e4d5-44ff-9923-ef49fc8fe03b",
+   "metadata": {},
+   "source": [
+    "To check that everything worked as expected, you can run `ls -lh ./seed-scripts/fhir-data/ | wc -l`, which should return a value of `1282`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "437822b7-a9e1-4b35-b172-1a518c463709",
+   "metadata": {},
+   "source": [
+    "In order to launch the ECR Viewer, you should simply need to run `docker compose up` from the `phdi/containers/ecr-viewer` directory. However, it should also fail. Seeding the database will fail because not all of the data successfully converted, so there are erroneous files.\n",
```

Comment:
It might be helpful to add the previous [note](https://skylight-hq.slack.com/archives/C06K10M6YF5/p1709160252459759?thread_ts=1709147029.505309&cid=C06K10M6YF5) about error handling to get things up and running, so that the code lives alongside the notebook rather than in Slack.

Reply:
Out of scope for this PR: How are failures in eCR -> FHIR conversion currently handled? How might these failures impact the RL service?

Reply:
Great question! My very initial testing implies that errors are handled silently for the most part, and generate a blank `.json` file. But more rigorous testing should be explored.

Reply:
With the de-identified LAC data, I noticed a couple error results.  Most of them (~40) we're returned from the FHIR API will a payload of "null", however 2 of them came back with empty payloads.